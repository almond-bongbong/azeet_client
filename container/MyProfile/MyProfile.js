import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image, { ThumbnailStyle } from 'components/Thumbnail/Thumbnail';
import Button from 'components/Button/Button';
import { useDispatch } from 'react-redux';
import { confirmActions } from 'store/modules/confirm';
import { authActions } from '../../store/modules/auth';

export const MyProfileStyle = styled.div`
  padding: 50px;
  background-color: #28282d;
  box-shadow: 1px 1px 5px 1px rgba(0,0,0,0.3);
  text-align: center;
  
  ${ThumbnailStyle} {
    overflow: hidden;
    width: 120px;
    height: 120px;
    margin: 0 auto;
    border-radius: 50%;
  }
  
  .name {
    margin: 10px 0;
    text-align: center;
  }
`;

const MyProfile = ({ user }) => {
  const { nickname, profileImage } = user;
  const dispatch = useDispatch();
  const handleLogout = useCallback(async () => {
    if (await dispatch(confirmActions.confirm('정말 로그아웃 하시겠습니까?'))) {
      dispatch(authActions.logout());
    }
  }, [dispatch]);

  return (
    <MyProfileStyle>
      <Image url={profileImage} />
      <div className="name">{nickname}</div>
      <Button height={30} onClick={handleLogout} theme="gray" text="로그아웃" />
    </MyProfileStyle>
  );
};

MyProfile.propTypes = {
  user: PropTypes.shape({
    nickname: PropTypes.string,
    profileImage: PropTypes.string,
  }),
};

MyProfile.defaultProps = {
  user: {},
};

export default MyProfile;
