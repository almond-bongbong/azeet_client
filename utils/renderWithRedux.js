import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configStore from 'store/configure';

function renderWithRedux(ui, initialState) {
  const store = configStore(initialState);
  const utils = render(
    <Provider store={store}>
      {ui}
    </Provider>,
  );
  return {
    ...utils,
    store,
  };
}

export default renderWithRedux;
