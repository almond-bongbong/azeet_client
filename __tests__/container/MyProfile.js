import React from 'react';
import renderWithRedux from 'utils/renderWithRedux';
import { fireEvent, waitForElement } from '@testing-library/react';
import MyProfile from 'container/MyProfile';

describe('MyProfile', () => {
  const mockUser = {
    nickname: 'cmlee',
    profileImage: 'https://placeimg.com/600/300/any',
  };

  it('matches snapshot', () => {
    const utils = renderWithRedux(<MyProfile />);
    expect(utils.container).toMatchSnapshot();
  });

  it('login form', async () => {
    const { getByText } = renderWithRedux(<MyProfile user={mockUser} />);
    getByText('cmlee');
    const btnLogout = getByText('로그아웃');

    fireEvent.click(btnLogout);

    const confirm = await waitForElement(() => getByText('정말 로그아웃 하시겠습니까?'));
    const btnYes = await waitForElement(() => getByText('확인'));

    expect(confirm).not.toBeNull();
    expect(btnYes).not.toBeNull();
  });
});
