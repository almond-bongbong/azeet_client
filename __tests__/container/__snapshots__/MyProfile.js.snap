// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile matches snapshot 1`] = `
<div>
  <div
    class="MyProfile__MyProfileStyle-sc-46uf30-0 fpKnAn"
  >
    <div
      class="Thumbnail__ThumbnailStyle-sc-10cu7dh-0 fKKAJA"
    >
      <img
        alt="thumbnail"
        class="image"
      />
    </div>
    <div
      class="name"
    />
    <div
      class="Button__ButtonStyle-sc-1myzjdo-0 fdfBKF"
      data-testid="button-wrap"
      height="30"
      width="auto"
    >
      <button
        type="button"
      >
        <span>
          로그아웃
        </span>
      </button>
    </div>
  </div>
</div>
`;
